Rebrush Elements in task subpages

zuhrasofyan opened the pull request:
Apply styling for elements in task subpages:

### Elements re-brushed:
- Heading in view/edit task
- Star rating
- Numbering
- Dropdown button
- Dropdown comment
- Button group
- Form group

### Some clean-up:
- Replace inline styling of button icons with bootstrap's text-color classes
- Combining and small fixes on CSS
- Reducing the number of different color implemented in elements

### Implement similar element styling in other pages

### Here is the example of implemented element changes:
Before: 
<img width="1332" alt="Screenshot 2024-04-22 at 21 23 39" src="https://github.com/openHPI/codeharbor/assets/4398374/563e31d2-6a15-4524-8fd0-6e9a0af3fcd5">

![screencapture-codeharbor-openhpi-de-tasks-851-2024-04-16-23_26_59](https://github.com/openHPI/codeharbor/assets/4398374/d8047d9f-de4f-429e-98cc-8e6f0229e390)

After:
<img width="1349" alt="Screenshot 2024-04-22 at 21 13 59" src="https://github.com/openHPI/codeharbor/assets/4398374/6bb999da-b66f-4de4-92d9-cf0b16348e40">

![screencapture-127-0-0-1-7500-tasks-1-2024-04-16-23_26_09](https://github.com/openHPI/codeharbor/assets/4398374/a84cf1af-bba9-4881-ba9e-c4121cdb083e)

kkoehn reviewed (changes requested):
Thanks for the design-rework. It looks really good and modern, but I'm sad to see most of the colors go.

Please have a look at the following points:
Your screenshot from the task-index looks good, but for me it looks like this (Firefox):
![image](https://github.com/openHPI/codeharbor/assets/24592181/856b6a24-2449-48bc-ad98-f988dbc064dd)
the vertical alignment of the rating-stars seems to be broken

For the group-show the (disabled) "Leave Group" also looks weird (at least the right border is missing but the white background also doesn't really fit)
![image](https://github.com/openHPI/codeharbor/assets/24592181/a46bfdc2-6319-4a96-af01-3c65384f6f11)

Task-view:
inconsistent font-weight for "Files:"
![image](https://github.com/openHPI/codeharbor/assets/24592181/8a0b4cc8-8ea1-4d57-822d-32b83f5e80c6)

zuhrasofyan commented:
For the cosmetics improvement you [suggested](https://github.com/openHPI/codeharbor/pull/1359#pullrequestreview-2048638364), unfortunately for the sake of simplification and not to have too many color variation, I tried to streamline and reduce too many color being used :/. Anyway, here are some improvement based on your review:
1) I adjusted the font size to fit the border alignment. Here the star-rating in firefox:
<img width="847" alt="Screenshot 2024-05-23 at 12 17 16" src="https://github.com/openHPI/codeharbor/assets/4398374/36dc23e5-9fd9-4f44-8dae-f7ca496f801e">

2) I've changed the color and border for disabled button in button group. Let me know if you have a better color match idea:
<img width="603" alt="Screenshot 2024-05-23 at 12 17 37" src="https://github.com/openHPI/codeharbor/assets/4398374/72c918d6-37f1-4132-b411-41bcac08992c">

3) Thanks for spotting that. It was the legacy code, and I forgot to adjust the similar styling.

zuhrasofyan commented on app/views/account_links/show.html.slim:
```diff
@@ -1,32 +1,35 @@
+    .col-md-12.pb-2.border-bottom
+      .std-heading
+        i.fa-solid.fa-user
+        = " #{AccountLink.model_name.human}: #{@account_link.name}"
```
Resolved

zuhrasofyan commented on app/views/tasks/_tasks.html.slim:
```diff
@@ -3,67 +3,82 @@
+                  i.fas.fa-lock-open
+                - else
+                  i.fas.fa-lock
+              / span.visibility data-bs-toggle="tooltip" data-bs-placement="top" title="#{t('tasks.visibility.label')}"
+              /   = t("tasks.visibility.#{task.access_level}")
```
We don't. I will delete those lines

kkoehn commented on app/assets/stylesheets/exercises.css.scss:
```diff
@@ -138,30 +125,62 @@ $exercise-color: #006600;
 .fa-regular.fa-star,
 .fa-regular.fa-star-half-stroke {
-  color: #b79a00;
+  color: $exercise-color;
```
since we moved from exercise to task `$task-color` would fit better, I think

kkoehn commented on app/views/tasks/_tasks.html.slim:
```diff
@@ -3,67 +3,80 @@
+            = truncate(task.title, length: 100)
+            - if @visibility == :owner
+              span.visibility.align-middle
+                - if task.access_level == 'public'
```
```suggestion
                - if task.access_level_public?
```

kkoehn reviewed (commented):
Thanks for incorporating the suggestions.
I found a few more minor things. 

While testing the new design, I also stumbled over the following thing in `collection/show`:
![image](https://github.com/openHPI/codeharbor/assets/24592181/7a8123fc-0c74-454d-9dd4-84982c85d2d6)
While I don't think you touched it, could you still fix it?

MrSerth commented:
The `Collections#show` route looks weird in German:

<img width="1319" alt="Bildschirmfoto 2024-05-26 um 12 24 15" src="https://github.com/openHPI/codeharbor/assets/7300329/06b7cd64-352a-4025-9dee-bca445e52a61">

First, the action buttons are really too long with non-unified line breaks (see the "Back" button: It's not centered as all other text). Then, there are some elements still using colors, whereas others lost it (check the "statistics" about members, invitations, the title, ...).

MrSerth commented:
Regarding the colors: I still feel that we should either go with some colored icons or none at all. What do you think about this mix?

![Bildschirmfoto 2024-05-29 um 12 06 47](https://github.com/openHPI/codeharbor/assets/7300329/c61e29c1-8429-4300-a988-573101ddf46c)

zuhrasofyan commented:
The star in the title will disappear once we remove it from the favorite, so I personally think the star color should be distinguished, but not the rest.

zuhrasofyan commented:
Fun fact: I am indeed literally partial color blind. So, this one is totally on me. I agree with your point and gladly will fix it 😄

MrSerth commented:
Ah, I didn't know, no worries.

Just to be sure: The star in the title is currently light grey, the folder icon is dark grey/black (similar to the title text). Is this what you mean with "the star color should be distinguished"?

zuhrasofyan commented:
Yes :)

kkoehn reviewed (approved):
Looks good to me now! Thanks for incorporating the feedback!
